### PR TITLE
Observation Summary Table

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
Adds the observation summary table view when not selecting an observation.

There's still some work to be done for the table:

- Some columns are not supported yet, this will have to be picked up later when the fields are added
- Some target-specific columns are not added yet as grouping should be done for those
- Elevation view

### CODE REVIEW

File: explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+      ),
+      // TableStore(props.userId, TableId.ObservationsSummary, cols)
+      // )
+    )
+    .render { (props, _, _, _, table) =>
+      <.div(
+        props.renderInTitle(
+          React.Fragment(
+            <.span(), // Push column selector to right
+            <.span(ExploreStyles.TitleSelectColumns)(
```
Comment (hugo-vrijswijk) [inline explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala:176]: Clicking a row should navigate to the observation selection

**Comment by cquiroz** [comment]
Uh, congratulations on your first PR!

File: explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+          // TODO: RAColumnId
+          // TODO: DecColumnId
+          // TODO: TimingColumnId
+          // TODO: SEDColumnId
+          // TODO: ChargedTimeColumnId
+        )
+    )
+    // Rows
+    .useMemoBy((props, _, _) => props.observations.get)((_, _, _) => _.toList)
+    // TODO: useReactTableWithStateStoreBy
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala:158]: Do we need to add the key to the DB in order to store the prefs?

**Review by rpiaggio — APPROVED** [review]
Looks great, thank you!

**Comment by rpiaggio** [comment]
There's a `fixImports` alias in explore to run `scalafix OrganizeImports` throughout.

There's also a `prePR` task in all projects (comes from `sbt-lucuma`) that cleans the project, rebuilds, regenerates workflows and runs all linters and MIMA checks.

File: explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala (hunk 2 above)
Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala:158]: Yes

**Review by cquiroz — COMMENTED** [review]
This seems good, we can do the update of the missing table id in another PR
Thanks and congratulation for your first PR

File: explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    )
+
+object ObsSummaryTable extends TableHooks:
+  private type Props = ObsSummaryTable
+
+  private val ColDef = ColumnDef[ObsSummaryWithTitleConstraintsAndConf]
+
+  private val GroupsColumnId          = ColumnId("groups")
+  private val ObservationIdColumnId   = ColumnId("observation_id")
+  private val ValidationCheckColumnId = ColumnId("validation_check")
```
[the remark this answers is not in the document]
Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/observationtree/ObsSummaryTable.scala:72]: afbeelding
It's these checks

**Pull request merged**